custom_components/moonraker/config_flow.py

```diff
@@ -26,17 +26,22 @@ async def async_step_user(self, user_input=None):
 
         if user_input is not None:
             valid = await self._test_connection(user_input[CONF_URL])
-            if valid:
-                return self.async_create_entry(
-                    title=DOMAIN, data=user_input
-                )  # changer DOMAIN pour name
-            self._errors["base"] = "error"
+            if not valid:
+                self._errors["base"] = "error"
+                return await self._show_config_form(user_input)
 
-            return await self._show_config_form(user_input)
+            valid = await self._test_port(user_input[CONF_PORT])
+            if not valid:
+                self._errors[CONF_PORT] = "port_error"
```

Comment:
Feels weird that this one use a const and not `self._errors["base"] = "error"`

Reply:
base mean that the error is on top of the config windows and if it's equal the name of a fields the error will display at that fields in the UI.

I think it's the other one that should use CONF_URL in fact, i'll add this